### DISCUSSION TITLE
Can't upload a file to S3. Can't see the problem.

### DISCUSSION THREAD

**Discussion by qiqetes** [serverpod/serverpod#2651] [Q&A]
```dart
// CLIENT CODE
String? uploadDescription =
    await client.X.getUploadImageDescription();

if (uploadDescription == null) return;

final image = File(state.compressedImagePath!);
print("Attempting upload: ${image.path}");
final Stream<List<int>> stream = image.openRead();

final fileUploader = FileUploader(uploadDescription);
bool uploaded =
    await fileUploader.upload(stream, image.readAsBytesSync().length);
print("Was uploaded $uploaded"); // Always FALSE, nothing on the S3
// bool uploaded = await client.X.verifyUpload();
```

I get the uploadDescription, looks ok, the image that I'm trying to upload is 8kb. I wish it wasn't a true/false but an Exception with some message what comes from `upload` method. I don't know what I'm doing wrong.

I think I have followed the docs very meticulously, still can't upload there. I have set the S3 private with a CloudFront, and this at the server side:
```dart
return await session.storage
      .createDirectFileUploadDescription(storageId: "private", path: "noslashatstart/whatever/x"); 
```

In server.dart:
```dart
 pod.addCloudStorage(s3.S3CloudStorage(
    serverpod: pod,
    storageId: 'private',
    public: false,
    region: 'us-west-1',
    bucket: 'xxx',
    publicHost: 'xxxx.cloudfront.net',
  ));
  ```

**Comment by qiqetes** [comment]
After debugging I get the error:
`<Error><Code>AccessControlListNotSupported</Code><Message>The bucket does not allow ACLs</Message>` which I don't understand very well, but I guess it's my fault.

**Comment by qiqetes** [comment]
The filedescription comes from S3, how does it try to upload with ACL if ACL is not allowed in my bucket?

**Reply by Isakdl** [reply]
We recently fixed this for new projects, you can update your terraform script to include the ObjectOwner.

see the PR: https://github.com/serverpod/serverpod/pull/2650/files#diff-0668e588d67b06c236f18fe0e87abc19c80525f5338661b61a06a8b782d7da15
